### DISCUSSION TITLE
Drop Markdown Support

### DISCUSSION THREAD

**Discussion by shadowtime2000** [alephjs/aleph.js#91] [Ideas]
I feel like automatically converting markdown into a React component may not be that good of an idea. First of all, it makes AlephJS more opinionated than it needs to be. Second, some people (like me) want to separate content from code. I wouldn't want to create a blog post in `pages/blog/my-blog-post.md` but rather in `posts/my-blog-post.md` with `pages/blog/$post` automatically loading it.

**Comment by mohsenkhanpour** [comment]
I think markdown support is a great idea. Aleph docs are written in markdown.
It would be a great idea to support .mdx and .html too.
I also agree with @shadowtime2000  `pages` directory seems to be redundant. I think it would be a good idea to add a input option to cli like `aleph --input=. --output=_dist --ext=md,tsx` and also to aleph.config.js to handle the input directory.
Also it would be a good idea to choose layout for markdown pages in frontmatter.

**Reply by ije** [reply]
@shadowtime2000 @mohsenkhanpour i alreay dropped it in https://github.com/alephjs/aleph.js/pull/98/commits/8162d2f56652b6f16df8b40aef56c69019fb8c77 , and i added a new `ServePlugin` type to allow us to add custom page modules like `markdown`:

```ts
export default {
    type: 'server'
    onInit(app: ServerApplication): {
        for await (const { path: p, } of walk(path.join(app.workingDir, 'docs'), { exts: ['.md', '.markdown'] })) {
            app.addPageModule('/docs/' + p.split('/docs/')[1].replace(/\.md$/, ''), toJs(Deno.readTextFileSync(p)))
        }
        // todo: watch md files...
    }
}
```
